app/lib/screens/study/tasks/task_screen.dart

```diff
@@ -37,16 +37,33 @@ class _TaskScreenState extends State<TaskScreen> {
   Widget _buildTask() {
     switch (taskInstance.task) {
       case CheckmarkTask checkmarkTask:
-        return CheckmarkTaskWidget(
-          task: checkmarkTask,
-          key: UniqueKey(),
-          completionPeriod: taskInstance.completionPeriod,
+        return SingleChildScrollView(
+          child: SizedBox(
+            width: MediaQuery.of(context).size.width,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: [
+                HtmlText(taskInstance.task.header, centered: true),
+                const SizedBox(height: 20),
+                CheckmarkTaskWidget(
+                  task: checkmarkTask,
+                  key: UniqueKey(),
+                  completionPeriod: taskInstance.completionPeriod,
+                )
+              ],
+            ),
+          ),
         );
       case QuestionnaireTask questionnaireTask:
-        return QuestionnaireTaskWidget(
-          task: questionnaireTask,
-          key: UniqueKey(),
-          completionPeriod: taskInstance.completionPeriod,
+        return Flex(
+          direction: Axis.vertical,
+          children: [
+            QuestionnaireTaskWidget(
```

Comment:
Why do you use a Flex with only one child.

Reply:
QuestionnaireTaskWidget is using Expanded that needs to be encapsulated. I refactored the part to make it more clear.